### PR TITLE
Styled section as styled component

### LINKED ISSUE #415: Use styled-components for styling StyledSection
We style the StyledSection component using SCSS/CSS. To be able to start using it in the new content analysis rendering/Gutenberg we need it to use styled-components.

### PR DESCRIPTION
**Pull request by afercia**
For the "Insights" section on Premium, needs:
https://github.com/Yoast/wordpress-seo-premium/pull/1567
https://github.com/Yoast/yoast-premium-components/pull/43

- extract the heading in a separate component, with test
- makes the Section heading optional, rename props
- introduces a StyledSection component with a Heading an optional icon
- introduces a SvgIcon component

To check: are the Section and StyledSection currently used in other components? Haven't found other usages, any help in double checking very welcome.

To test the standalone version, use this code in App.js. Test changing props: heading level, heading color, icon, icon color, etc.
```
import React from "react";
import styled from "styled-components";

import StyledSection from "./forms/StyledSection/StyledSection";
import Section from "./forms/Section";

const Container = styled.div`
	max-width: 800px;
	margin: 0 auto;
`;

class App extends React.Component {

	constructor() {
		super();
	}

	render() {
		return (
			<Container>
				<StyledSection
					headingText="Insights. Hello, this is a Styled Section heading."
					headingColor="red"
					headingLevel={ 2 }
					headingIcon="file-text"
					headingIconColor="blue"
					headingIconSize="16"
				>
					<p>Keyword Suggestions</p>
					<p>Hello, this is additional content</p>
				</StyledSection>
				<br />
				<Section
					headingText="Hello, this is a normal Section heading"
					headingLevel={ 3 }
				>
					<p>Normal Section content</p>
				</Section>
			</Container>
		);
	}
}

export default App;

```

Testing the Premium "Insights" section is a bit more complicated. What I've done:
- switch Premium and `yoast-premium-components` to the branches related to the PRs above
- on `yoast-premium-components`, yarn link `yoast-components`
- on premium, inside the `premium/` directory, yarn link `yoast-components`
- on premium, inside the `premium/` directory, build the JS

This should be enough to see the "Insights" section updated.

Note: I've tested the standalone version on Windows, IE 11 and Edge, everything looks fine. I can't test Premium on Windows because Git on Windows hates me.

Fixes #415

### CODE REVIEW

**Comment by afercia** [comment]
See https://github.com/Yoast/wordpress-seo-premium/pull/1567
(seems waffle hasn't linked it)

File: forms/StyledSection/StyledSection.js [hunk 1]
```diff
@@ -1,32 +1,92 @@
...
+				{ props.headingIcon &&
+					<StyledIcon
+						icon={ icons[ `${ props.headingIcon }` ] }
+						color={ props.headingIconColor }
+						size={ props.headingIconSize }
+					/>
+				}
+				{ props.headingText }
+			</StyledHeading>
+			{ props.sectionContent }
```
Comment (Xyfi) [inline forms/StyledSection/StyledSection.js]: This prop is unnecessary. I think all "regular" content should be passed through `props.children`. Allowing developers to achieve the same result with different approaches will cause inconsistent code.

File: forms/StyledSection/StyledSection.js [hunk 2]
```diff
@@ -1,32 +1,92 @@
...
 import Section from "../Section";
+import Heading from "../../composites/basic/Heading";
+import colors from "../../style-guide/colors.json";
+import { rgba } from "../../style-guide/helpers";
+import { Icon } from "../../composites/Plugin/Shared/components/Icon";
+import * as icons from "../../style-guide/svg";
+
+const StyledHeading = styled( Heading )``;
+
+const StyledIcon = styled( Icon )``;
```
Comment (Xyfi) [inline forms/StyledSection/StyledSection.js]: Use Icon if no additional styles are needed.

File: forms/StyledSection/StyledSection.js [hunk 3]
```diff
@@ -1,32 +1,92 @@
...
+import styled from "styled-components";
 
 import Section from "../Section";
+import Heading from "../../composites/basic/Heading";
+import colors from "../../style-guide/colors.json";
+import { rgba } from "../../style-guide/helpers";
+import { Icon } from "../../composites/Plugin/Shared/components/Icon";
+import * as icons from "../../style-guide/svg";
+
+const StyledHeading = styled( Heading )``;
```
Comment (Xyfi) [inline forms/StyledSection/StyledSection.js]: Use Heading if no additional styles are needed.

Comment (afercia) [reply forms/StyledSection/StyledSection.js]: As far as I know, needs to be wrapped in a styled components to be able to do `& ${ StyledHeading } { ...` later.

File: forms/StyledSection/StyledSection.js (hunk 2 above)
Comment (afercia) [reply forms/StyledSection/StyledSection.js]: See above.

File: forms/StyledSection/StyledSection.js (hunk 1 above)
Comment (afercia) [reply forms/StyledSection/StyledSection.js]: I'd agree, but this is the way content is currently passed on Premium in `premium/assets/js/src/keywordSuggestions/KeywordSuggestions.js` so we should make changes there too.

**Comment by afercia** [comment]
See updated test instructions above.

**Comment by afercia** [comment]
Just realized this should wait, as we can't delete `forms/StyledSection/StyledSection.scss` because it is imported in the plugin in `metabox.scss`

**Comment by afercia** [comment]
Added the `key` and `list` icons currently used in the metabox for the Focus keyword and the Analysis heading. They will be needed when these headings will be styled components. See comment in the plugin `metabox.scss`
`// TODO: Move this to the yoast-components repository`

**Comment by igorschoester** [comment]
Acceptance 👍

**Pull request merged**